x/group/types.go

```diff
@@ -113,6 +108,68 @@ func (p *ThresholdDecisionPolicy) Validate(g GroupInfo) error {
 	return nil
 }
 
+// Implements DecisionPolicy Interface
+var _ DecisionPolicy = &PercentageDecisionPolicy{}
+
+// NewPercentageDecisionPolicy creates a new percentage DecisionPolicy
+func NewPercentageDecisionPolicy(percentage string, timeout time.Duration) DecisionPolicy {
+	return &PercentageDecisionPolicy{percentage, timeout}
+}
+
+func (p PercentageDecisionPolicy) ValidateBasic() error {
+	percentage, err := math.NewPositiveDecFromString(p.Percentage)
+	if err != nil {
+		return sdkerrors.Wrap(err, "percentage threshold")
+	}
+	if percentage.Cmp(math.NewDecFromInt64(1)) == 1 {
+		return sdkerrors.Wrap(errors.ErrInvalid, "percentage must be > 0 and <= 1")
+	}
+
+	timeout := p.Timeout
+	if timeout <= time.Nanosecond {
+		return sdkerrors.Wrap(errors.ErrInvalid, "timeout")
+	}
+	return nil
+}
+
+func (p *PercentageDecisionPolicy) Validate(g GroupInfo) error {
+	return nil
+}
+
+// Allow allows a proposal to pass when the tally of yes votes equals or exceeds the percentage threshold before the timeout.
+func (p PercentageDecisionPolicy) Allow(tally Tally, totalPower string, votingDuration time.Duration) (DecisionPolicyResult, error) {
+	timeout := p.Timeout
+	if timeout <= votingDuration {
+		return DecisionPolicyResult{Allow: false, Final: true}, nil
+	}
+
+	percentage, err := math.NewPositiveDecFromString(p.Percentage)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	yesCount, err := math.NewNonNegativeDecFromString(tally.YesCount)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+	totalPowerDec, err := math.NewNonNegativeDecFromString(totalPower)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+
+	yesPercentage, err := yesCount.Quo(totalPowerDec)
+	if err != nil {
+		return DecisionPolicyResult{}, err
+	}
+
+	if yesPercentage.Cmp(percentage) >= 0 {
+		return DecisionPolicyResult{Allow: true, Final: true}, nil
+	} else if yesPercentage.Cmp(percentage) < 0 {
+		return DecisionPolicyResult{Allow: false, Final: true}, nil
```

Comment:
please have a look at the existing implementation in `ThresholdDecisionPolicy`
similarly, we can say that a proposal doesn't pass as soon as (yes + undecided) / totalPowerDec < percentage